tests/integration_test.rs

```diff
@@ -80,8 +80,93 @@ fn identity_map() {
              excusable);
 }
 
+#[test]
+#[cfg(not(target_os = "macos"))]
+fn with_functions() {
+    // Ignored on macOS since atos doesn't support -f
+
+    let target = env::current_exe().unwrap();
+    let mut debug = target.clone();
+
+    // On macOS, symbols are kept in a separate .dSYM file
+    if cfg!(target_os = "macos") {
+        let fname = debug.file_name().unwrap().to_string_lossy().into_owned();
+        debug.set_file_name(format!("{}.dSYM", fname));
+        debug.push("Contents");
+        debug.push("Resources");
+        debug.push("DWARF");
+        debug.push(fname);
+    }
+
+    // Parse the debug symbols using "our" addr2line
+    let ours = addr2line::Mapping::with_functions(&debug).unwrap();
+
+    // Spin up the "real" addr2line
+    let mut theirs = spawn_oracle(target.as_path(), &["-f"]);
+
+    // Find some addresses to test using nm (we'll later filter to just text section symbols)
+    let mut theirs_in = theirs.stdin.take().unwrap();
+    let mappings = get_test_addresses(target.as_path(), &mut theirs_in);
+    drop(theirs_in);
+
+    // Go through addresses one by one, and check that we get the same answer as addr2line does.
+    use std::io::BufReader;
+    let mut theirs_out = BufReader::new(theirs.stdout.take().unwrap()).lines();
+    let mut all = 0;
+    let mut excusable = 0;
+    let mut func_hits = 0;
+    let mut got = 0;
+    for addr in mappings {
+        // Read the oracle ouput
+        let function = theirs_out.next().unwrap().unwrap();
+        let oracle = theirs_out.next().unwrap().unwrap();
+        let oracle = canonicalize_oracle_output(&*oracle);
+
+        // Does our answer match?
+        if let Some((mut file, lineno, func)) = ours.locate(addr) {
+            // We dared to guess -- did we give the right answer?
+            if cfg!(target_os = "macos") {
+                // atos doesn't include the full path, just file name
+                let f = path::PathBuf::from(&file);
+                let f = f.file_name().unwrap().to_string_lossy().into_owned();
+                file = f;
+            }
+
+            assert_eq!(oracle.0, Some(&*file));
+            assert_eq!(oracle.1, Some(lineno));
+
+            if let Some(func) = func {
+                // We even tried to guess the function!
+                assert_eq!(function, func);
```

Comment:
As far as I can tell, this is the failing test line, which seems to be failing because of potentially different demangling:
`_ZN100_$LT$core..slice..Iter$LT$$u27$a$C$$u20$T$GT$$u20$as$u20$core..iter..traits..DoubleEndedIterator$GT$9next_back17h965518d3b70e08caE` looks quite similar to`_ZN4core5slice8{{impl}}13next_back<u8>E`.

But it's probably not just demangling; though similar enough that it's suspicious.